framework/db/Query.php

```diff
@@ -553,6 +553,8 @@ public function andWhere($condition, $params = [])
     {
         if ($this->where === null) {
             $this->where = $condition;
+        } elseif (is_array($this->where) && isset($this->where[0]) && strtoupper($this->where[0]) == 'AND') {
```

Comment:
`strncasecmp($this->where[0], 'AND', 3)` could be used instead of `strtoupper` and comparison.